frappe/model/delete_doc.py

```diff
@@ -302,13 +303,12 @@ def check_if_doc_is_dynamically_linked(doc, method="Delete"):
 				refdoc.get(df.options) == doc.doctype
 				and refdoc.get(df.fieldname) == doc.name
 				and (
-					(method == "Delete" and refdoc.docstatus < 2)
-					or (method == "Cancel" and refdoc.docstatus == 1)
+					# linked to an non-cancelled doc when deleting
+					(method == "Delete" and not DocStatus(refdoc.docstatus).is_cancelled())
+					# linked to a submitted doc when cancelling
```

Comment:
same here - also not a fan of the comments added inbetween the condition

Reply:
I moved the comments there so that it is clear what they are referring to. At their original location, it wasn't immediately obvious to me that they are describing these particular conditions.

But I guess that's a matter of taste. 🤷🏻‍♂️